internal/controllers/addon/addon_instance_reconciler.go

```diff
@@ -61,31 +64,48 @@ func (r *addonInstanceReconciler) ensureAddonInstance(
 	}
 
 	if err := controllerutil.SetControllerReference(addon, desiredAddonInstance, r.scheme); err != nil {
-		return fmt.Errorf("setting controller reference: %w", err)
+		return resultRetry, fmt.Errorf("setting controller reference: %w", err)
 	}
 
-	return r.reconcileAddonInstance(ctx, desiredAddonInstance)
+	return r.reconcileAddonInstance(ctx, desiredAddonInstance, addon)
 }
 
 // Reconciles the reality to have the desired AddonInstance resource by creating it if it does not exist,
 // or updating if it exists with a different spec.
 func (r *addonInstanceReconciler) reconcileAddonInstance(
-	ctx context.Context, desiredAddonInstance *addonsv1alpha1.AddonInstance) error {
+	ctx context.Context, desiredAddonInstance *addonsv1alpha1.AddonInstance, addon *addonsv1alpha1.Addon) (requeueResult, error) {
 	currentAddonInstance := &addonsv1alpha1.AddonInstance{}
 	err := r.client.Get(ctx, client.ObjectKeyFromObject(desiredAddonInstance), currentAddonInstance)
 	if errors.IsNotFound(err) {
-		return r.client.Create(ctx, desiredAddonInstance)
+		return resultNil, r.client.Create(ctx, desiredAddonInstance)
 	}
 	if err != nil {
-		return fmt.Errorf("getting AddonInstance: %w", err)
+		return resultRetry, fmt.Errorf("getting AddonInstance: %w", err)
 	}
+
+	// If addon's operator choose to onboard addon instance into their workflow.
+	// Based on the addon instance 'Installed' condition, influence the addon's 'Available' and 'Installed' conditions.
+	if installedCond := meta.FindStatusCondition(
+		currentAddonInstance.Status.Conditions,
+		addonsv1alpha1.AddonInstanceConditionInstalled.String(),
+	); installedCond != nil {
+		if meta.IsStatusConditionFalse(currentAddonInstance.Status.Conditions, addonsv1alpha1.AddonInstanceConditionInstalled.String()) {
```

Comment:
We have to careful about transitioning the installed condition from true to false. IE, we have to get some agreement from the tenants that this installed condition is not overloaded to become a proxy "available" condition. 
The installed condition should transition only once from true to false.

Reply:
Thanks for pointing this out, now that I understand, I agree these conditions shouldn't be interdependent.